src/System.Numerics.Tensors/System/Numerics/Tensor.cs

```diff
@@ -1431,6 +1430,54 @@ public virtual CompressedSparseTensor<T> ToCompressedSparseTensor()
 
         #endregion
 
+        #region Slice
+
+        public Tensor<T> Slice(params Range[] ranges)
+        {
+            return Slice(new ReadOnlySpan<Range>(ranges));
+        }
+
+        public Tensor<T> Slice(ReadOnlySpan<Range> ranges)
+        {
+            if (ranges.Length != Rank)
+            {
+                throw new ArgumentException($"{nameof(ranges.Length)} of {nameof(ranges)} ({ranges.Length}) must match the {nameof(Rank)} of this {nameof(Tensor)} ({Rank}).");
+            }
+
+            Span<int> newDimensions = Rank < ArrayUtilities.StackallocMax ? stackalloc int[Rank] : new Span<int>(new int[Rank]);
+            Span<int> newStrides = Rank < ArrayUtilities.StackallocMax ? stackalloc int[Rank] : new Span<int>(new int[Rank]);
+            int offset = 0;
+            int newDimensionCount = 0;
+
+            // construct new dimensions and strides, compute offset as the linearization of the lower bounds
+            for (int i = 0; i < ranges.Length; i++)
+            {
+                var range = ranges[i];
+
+                // Range guarantees the Length is positive
+                Debug.Assert(range.Length > 0);
+
+                offset += range.Index * strides[i];
```

Comment:
potential integer overflow?

Reply:
Depending on how the clr is configured, e.g. what provisions have been alloted for by virtue of the fact the runtime exists and can be configured to prevent such... signed integer math is just unsigned  math with larger numbers....

Reply:
I've added a `checked` block to this line.